Contribute/how-to-write-use-markdown.md

```diff
@@ -383,10 +383,329 @@ You can see an example of selectors in action at the [Azure docs](https://docs.m
+
+        IEnumerable<int> scoreQuery =
+            from score in scores  
+            where score > 80  
+            select score;
+
+        // Execute the query.  
+        foreach (int i in scoreQuery)
+        {  
+            Console.Write(i + " ");
+        }
+    }  
+}
+```
+
+Code Snippet with Class Scaffolding Applied
+
+```md
+class NameOfClass {
+
+   public static void Main()
+    {
+        // Specify the data source.
+        int[] scores = new int[] { 97, 92, 81, 60 };
+
+        // Define the query expression.
+        IEnumerable<int> scoreQuery =
+            from score in scores
+            where score > 80
+            select score;
+
+        // Execute the query.
+        foreach (int i in scoreQuery)
+        {
+            Console.Write(i + " ");
+        }
+    }  
+}
+```
+
+The `try-dotnet-method` value applies method-level scaffolding to the code passed to the interactive component.
+
+```md
+:::code source="relative/path/source.cs" interactive="try-dotnet-method"::
+```
+
+Example:
+
+Code Snippet without Method Scaffolding Applied
+
+```md
+/*Print some string in C#*/
+
+Console.WriteLine("Hello C#.);
+
+Console.ReadKey();
+```
+
+Code Snippet with Method Scaffolding Applied
+
+```md
+public static void Main(string args[]) {
+
+/*Print some string in C#*/
+
+Console.WriteLine("Hello C#.);
+
+Console.ReadKey();
+
+}
+```
+
+#### Snippet syntax reference
+
+You can reference code snippets stored in your repo by using the specified code language. The
+content of the specified code path will be expanded and included in your file.
+
+There aren't restrictions on the folder structure of code snippets. You can manage the code snippets
+as normal source code.
+
+Syntax:
+
+```md
+:::code language="<language>" source="<path>" <attribute>="<attribute-value>":::
+```
+
+> [!IMPORTANT]
+> This syntax is a block Markdown extension. It must be used on its own line.
+
+- `<language>` (*optional*)
+  - Language of the code snippet. For more information, see the [Supported languages](#supported-languages)
+    section later in this article.
+
+- `<path>` (*mandatory*)
+  - Relative path in the file system that indicates the code snippet file to reference.
+
+- `<attribute>` and `<attribute-value>` (*optional*)
+  - Used together to specify how the code should be retrieved from the file:
+    - `range`: `1,3-5` A range of lines. This example includes lines 1, 3, 4, and 5.
+    - `id`: `snippet_Create` The ID of the snippet that needs to be inserted from the code file. This value cannot co-exist with range.
+    - `highlight`: `2-4,6` Range and/or numbers of lines that need to be highlighted in the generated code snippet. The numbering is relative to the code snippet itself, not the imported range.
+    - `interactive`: `cloudshell-powershell`, `cloudshell-bash`, `try-dotnet`, `try-dotnet-class`, `try-dotnet-method` String value determines what kinds of interactivity are enabled.
+
+#### Supported languages
```

Comment:
Somewhere near here, we should note that there are pairs of `language` and `interactive` that are compatible.  For example, if `interactive` is `try-dotnet`, `language` must be `csharp` (at least at this time).  Similarly, `cloudshell-powershell` would only work with `powershell` and `cloudshell-bash` would work only with `bash` as the language.

Reply:
I'll update to reflect this.